appcues/src/main/java/com/appcues/ui/extensions/TextPrimitiveExt.kt

```diff
@@ -0,0 +1,16 @@
+package com.appcues.ui.extensions
+
+import com.appcues.data.model.ExperiencePrimitive.TextPrimitive
+
+// helper to optionally adapt this Text to use the foreground color of a different Text, as an error state
+// used in form validation
+internal fun TextPrimitive.checkErrorStyle(showError: Boolean, errorLabel: TextPrimitive?): TextPrimitive {
+    val errorTint = if (showError) errorLabel?.style?.foregroundColor else null
```

Comment:
not sure if we could save some computing process by transforming this into a derivedStateOf that produces the TextPrimitive.

Reply:
I think this update will improve things https://github.com/appcues/appcues-android-sdk/pull/244/commits/0578f3c2f321e73f1054ab05de28665ae30a7290